### PR TITLE
Spec: Add _deleted, _spec_id, _partition metadata columns

### PR DESCRIPTION
**Pull request by aokolnychyi**
This PR updates the spec to include recently added metadata columns.

### CODE REVIEW

**Comment by aokolnychyi** [comment]
This is a follow-up to #2984 to address [this](https://github.com/apache/iceberg/pull/2984#discussion_r716254952) comment.

**Comment by aokolnychyi** [comment]
cc @rdblue @RussellSpitzer @jackye1995 @openinx @yyanyy

File: site/docs/spec.md [hunk 1]
```diff
@@ -232,6 +232,9 @@ The set of metadata columns is:
 |-----------------------------|---------------|-------------|
 | **`2147483646  _file`**     | `string`      | Path of the file in which a row is stored |
 | **`2147483645  _pos`**      | `long`        | Ordinal position of a row in the source data file |
+| **`2147483644  _deleted`**  | `boolean`     | Whether the row has been deleted |
+| **`2147483643  _spec_id`**  | `int`         | Spec ID used to track the file containing a row |
+| **`2147483642  _partition`** | `struct`     | Partition to which a row belongs to |
```
Comment (rdblue) [inline site/docs/spec.md:237]: Should just be "Partition to which a row belongs"

**Review by rdblue — APPROVED** [review]
Minor typo, but otherwise good.

File: site/docs/spec.md (hunk 1 above)
Comment (aokolnychyi) [reply site/docs/spec.md]: Fixed.

**Comment by aokolnychyi** [comment]
Thanks, @rdblue!

**Pull request merged**